libs/clients/syslumenn/src/lib/syslumenn.module.ts

```diff
@@ -0,0 +1,38 @@
+import { DynamicModule } from '@nestjs/common'
+import { SyslumennApi, Configuration } from '../../gen/fetch'
+import { createEnhancedFetch } from '@island.is/clients/middlewares'
+
+export interface SyslumennApiConfig {
+  url: string
+  username: string
+  password: string
+}
+
+export class SyslumennApiModule {
+  static register(config: SyslumennApiConfig): DynamicModule {
+    return {
+      module: SyslumennApiModule,
+      providers: [
+        {
+          provide: SyslumennApi,
+          useFactory: () => {
+            return new SyslumennApi(
```

Comment:
I wonder if we should be using the new config module thing

Reply:
Yes, that might be a good idea! I'll test it out